internal/infra/run.go

```diff
@@ -463,65 +463,60 @@ func putCloneDir(ctx context.Context, cli *client.Client, updater *Updater, dir
 	return nil
 }
 
+// Pull the image from the registry.
+// Always pull the image, even if it's already present locally.
+// This is because container image tags are mutable, and we always want to ensure we have the latest version.
 func pullImage(ctx context.Context, cli *client.Client, image string) error {
-	var inspect types.ImageInspect
-
-	// check if image exists locally
-	inspect, _, err := cli.ImageInspectWithRaw(ctx, image)
-
-	// pull image if necessary
-	if err != nil {
```

Comment:
This diff is horrible, but I essentially removed these 3 lines and decreased the indentation below.